react-native-spokestack.d.ts

```diff
@@ -0,0 +1,123 @@
+export = RNSpokestack;
+export as namespace RNSpokestack;
+
+declare namespace RNSpokestack {
+  function constructor(props?: object): void;
+
+  enum TraceLevel {
+    DEBUG = 10,
+    PERF = 20,
+    INFO = 30,
+    NONE = 100,
+  }
+
+  enum TTSFormat {
+    TEXT = 0,
+    SSML = 1,
+    SPEECHMARKDOWN = 2,
+  }
+
+  interface SpokestackRecognizeEvent {
+    transcript?: string;
+  }
+
+  interface SpokestackErrorEvent {
+    error?: string;
+  }
+
+  interface SpokestackTraceEvent {
+    message?: string;
+  }
+
+  interface SpokestackTTSEvent {
+    url?: string;
+  }
+
+  interface NLUSlot {
+    type: string;
+    value: string;
+  }
+
+  interface SpokestackNLUEvent {
+    result?: {
+      intent: string;
+      confidence: string;
+      slots: NLUSlot[];
+    };
+  }
+
+  interface SynthesizeOptions {
+    input: string;
+    format: TTSFormat;
+    voice: string;
+  }
+
+  type SpokestackEvent = SpokestackRecognizeEvent &
+    SpokestackErrorEvent &
+    SpokestackTraceEvent &
+    SpokestackTTSEvent &
+    SpokestackNLUEvent;
+
+  interface SpokestackConfig {
+    input: string;
+    stages: string[];
+    properties?: {
```

Comment:
There are several additional properties for wakeword configuration that are not listed here that may be set, see https://javadoc.io/doc/io.spokestack/spokestack-android/latest/io/spokestack/spokestack/wakeword/WakewordTrigger.html for a full list.